src/cswrap-core.c

```diff
@@ -366,6 +366,66 @@ static bool read_custom_opts(char **dst, const char *str)
     }
 }
 
+#if defined(PATH_TO_CSCLNG) || defined(PATH_TO_CSGCCA) \
+    || defined(PATH_TO_CSMATCH) || defined(PATH_TO_CSCPPC)
+/* if `path` starts with `prefix`, remove the prefix and the trailing ':' */
+static bool remove_path_prefix(char *path, const char *prefix)
+{
+    const size_t prefix_len = strlen(prefix);
+    if (strncmp(path, prefix, prefix_len))
+        // prefix not found
+        return false;
+
+    // skip all ':' that follow the prefix
+    const char *end = path + prefix_len;
+    while (*end == ':')
+        ++end;
+
+    // remove the prefix, including the trailing double dots
+    const size_t new_size = strlen(end) + /* NUL */ 1;
+    memmove(path, end, new_size);
+    return true;
+}
+
+/* if `path` starts with path to a known wrapper, remove the prefix */
+static bool remove_known_wrapper_from_path(char *path)
+{
+    bool removed = false;
+#ifdef PATH_TO_CSCLNG
+    if (remove_path_prefix(path, PATH_TO_CSCLNG))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSGCCA
+    if (remove_path_prefix(path, PATH_TO_CSGCCA))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSMATCH
+    if (remove_path_prefix(path, PATH_TO_CSMATCH))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSCPPC
+    if (remove_path_prefix(path, PATH_TO_CSCPPC))
+        removed = true;
+#endif
+    return removed;
+}
+
+/* remove paths to known wrappers from the beginning of $PATH */
+static void sanitize_path_for_analyzer(void)
+{
+    char *path = getenv("PATH");
+    if (!path || !*path)
+        return;
+
+    while (remove_known_wrapper_from_path(path))
+        ;
```

Comment:
nit: I'd say this looks better 
```suggestion
    while (remove_known_wrapper_from_path(path));
```

Reply:
I believe that writing the semicolon on a separate line is a common practice to highlight that loop body is empty.  This used to eliminate `-Wempty-body` warnings with both gcc and clang.  It does not seem to be needed in this case anymore but compilers still suggest it in other cases, such as:
```c
int foo(void)
{
    while (foo());
        return 0;
}
```
```
% clang -c xxx.c
xxx.c:3:18: warning: while loop has empty body [-Wempty-body]
    3 |     while (foo());
      |                  ^
xxx.c:3:18: note: put the semicolon on a separate line to silence this warning
1 warning generated.
```
I can also see this style being used in `gcc` itself: https://github.com/gcc-mirror/gcc/blob/a8e61cd71f0cda04d583722ca4c66301358b01e1/gcc/gcc.cc#L2722

Reply:
Alright, it's only a style issue anyway but to be honest, plain `;` on an empty line just looks disgusting.

Reply:
On the other hand, writing loop body (a null statement in this case) on a separate line is a common practice.

This coding style seems to be used consistently in `csutils`:
- https://github.com/csutils/cswrap/blob/b802b3e07727d7fbb2b662e64c5d7b13e0c63023/src/cswrap-util.c#L56
- https://github.com/csutils/csdiff/blob/e21d8ce96282f9cbc96bf8625319e9fdf546be85/src/lib/parser-gcc.cc#L827

I found one place where it is written on a single line but the doubled parentheses are just another way to suppress compiler warnings:
- https://github.com/csutils/cswrap/blob/b802b3e07727d7fbb2b662e64c5d7b13e0c63023/src/csexec-loader-common.c#L46

Reply:
Ok.